src/rust/ide/parser/tests/parsing.rs

```diff
@@ -60,30 +48,12 @@ impl Fixture {
         Fixture {parser:parser::Parser::new_or_panic()}
     }
 
-    /// Runs parser on given input, panics on any error.
-    fn parse(&mut self, program:&str) -> Ast {
-        println!("parsing {}", program);
-        let ast = self.parser.parse(program.into(), default()).unwrap();
-        assert_eq!(ast.shape().len(), program.len());
-        validate_spans(&ast);
-        assert_eq!(ast.repr(), program, "{:?}", ast);
-        ast
-    }
-
-    /// Program is expected to be single line module. The line's AST is
-    /// returned. Panics otherwise.
-    fn parse_line(&mut self, program:&str) -> Ast {
-        let ast  = self.parse(program);
-        let line = expect_single_line(&ast);
-        line.clone()
-    }
-
     /// Program is expected to be single line module. The line's Shape subtype
     /// is obtained and passed to `tester`.
     fn test_shape<T,F>(&mut self, program:&str, tester:F)
     where for<'t> &'t Shape<Ast>: TryInto<&'t T>,
                       F         : FnOnce(&T) -> () {
-        let ast   = self.parse_line(program);
+        let ast   = self.parser.parse_line(program).unwrap();
```

Comment:
Can we use expect instead of unwrap? It's also a note to self, because I used to do that but then I got relaxed about it.

Reply:
In this case I don't expect this to introduce any value:
1) code is test-only, so it will point us to the right place in case of failure and user will never see this
2) we use result, so we should get relevant error message from the Result Err contents. Using expect imho is much more useful with Option when you have no idea what went wrong and why it is empty.